src/main/java/bio/terra/cli/businessobject/resource/AzureStorageContainer.java

```diff
@@ -0,0 +1,109 @@
+    this.storageAccountId = configFromDisk.storageAccountId;
+    this.storageContainerName = configFromDisk.storageContainerName;
+  }
+
+  /** Deserialize an instance of the WSM client library object to the internal object. */
+  public AzureStorageContainer(ResourceDescription wsmObject) {
+    super(wsmObject.getMetadata());
+    this.resourceType = Type.AZURE_STORAGE_CONTAINER;
+    this.storageAccountId =
+        wsmObject.getResourceAttributes().getAzureStorageContainer().getStorageAccountId();
+    this.storageContainerName =
+        wsmObject.getResourceAttributes().getAzureStorageContainer().getStorageContainerName();
+  }
+
+  /** Deserialize an instance of the WSM client library create object to the internal object. */
+  public AzureStorageContainer(AzureStorageContainerResource wsmObject) {
+    super(wsmObject.getMetadata());
+    this.resourceType = Type.AZURE_STORAGE_CONTAINER;
+    this.storageAccountId = wsmObject.getAttributes().getStorageAccountId();
+    this.storageContainerName = wsmObject.getAttributes().getStorageContainerName();
+  }
+
+  /**
+   * Serialize the internal representation of the resource to the format for command input/output.
+   */
+  public UFAzureStorageContainer serializeToCommand() {
+    return new UFAzureStorageContainer(this);
+  }
+
+  /** Serialize the internal representation of the resource to the format for writing to disk. */
+  public PDAzureStorageContainer serializeToDisk() {
+    return new PDAzureStorageContainer(this);
+  }
+
+  /**
+   * Create an Azure storage container as a controlled resource in the workspace.
+   *
+   * @return the resource that was created
+   */
+  public static AzureStorageContainer createControlled(
+      CreateAzureStorageContainerParams createParams) {
+    validateResourceName(createParams.resourceFields.name);
+
+    // call WSM to create the resource
+    AzureStorageContainerResource createdResource =
+        WorkspaceManagerService.fromContext()
+            .createControlledAzureStorageContainer(
+                Context.requireWorkspace().getUuid(), createParams);
+    logger.info("Created Azure storage container: {}", createdResource);
+
+    // convert the WSM object to a CLI object
+    Context.requireWorkspace().listResourcesAndSync();
+    return new AzureStorageContainer(createdResource);
+  }
+
+  /** No-op: referenced Azure storage containers are not supported. */
+  protected void deleteReferenced() {}
+
+  /** Delete an Azure storage container controlled resource in the workspace. */
+  protected void deleteControlled() {
+    // call WSM to delete the resource
+    WorkspaceManagerService.fromContext()
+        .deleteControlledAzureStorageContainer(Context.requireWorkspace().getUuid(), id);
+  }
+
+  /** Resolve an Azure storage contaiber resource to a SAS token. */
+  public String resolve() {
+    return WorkspaceManagerService.fromContext()
+        .getAzureStorageContainerSasToken(Context.getWorkspace().get().getUuid(), id);
```

Comment:
FYI some thoughts on how this will look in the future -- For GCP resources, CLI and Axon UI manually constructs these strings ([example](https://cs.github.com/DataBiosphere/terra-cli/blob/main/src/main/java/bio/terra/cli/businessobject/resource/GcsBucket.java#L148)). I think it would be nice if WSM constructed them instead. Eg enumerateResources returns ResourceDescription, and ResourceDescription has a cloudPath field, which is set appropriately for Azure and GCP resources.

Reply:
Yeah, showing token with `resolve` seems a little odd. I think it's better to show token with describe. (Not list, we don't show details with list.)

Some examples of `resolve`:
- For buckets, `resolve` returns `gs://mybucket20220824-terra-vdevel-buoyant-plum-9341`
- For BQ table, `bigquery-public-data.human_genome_variants.1000_genomes_pedigree`

The idea is, people can use `resolve` in their notebook ([example](https://user-images.githubusercontent.com/10929390/180330386-29061df5-4de6-406f-90f8-e8321ddbbd53.png)) to get actual bucket name/BQ table name.

(The hope was, running a shell command is easier than writing FISS code.)

Reply:
Makes sense, I like including it with `describe`.